content/en/docs/reference/using-api/server-side-apply.md

```diff
@@ -578,8 +578,8 @@ Apply can send partially specified objects as YAML as the body of a `PATCH` requ
 to the URI of a resource.  When applying a configuration, you should always include all the
 fields that are important to the outcome (such as a desired state) that you want to define.
 
-All JSON messages are valid YAML. Some clients specify Server-Side Apply requests using YAML
-request bodies that are also valid JSON.
+All JSON messages are valid YAML. Some clients specify Server-Side Apply requests using JSON
+request bodies that are also valid YAML.
```

Comment:
`application/apply-patch+yaml` is always formally YAML, not JSON.

This change doesn't convey the nuance originally intended. You are welcome to write something new, but we shouldn't make the swap you're proposing here.

Reply:
JSON is YAML, but the reverse is not true. The sentence in the documentation currently states "Some clients specify Server-Side Apply requests using **YAML** request bodies that are also valid **JSON**", which is not true and the exact opposite. You can provide JSON (in addition to YAML) request bodies because **JSON is valid YAML**.

Reply:
A better wording might be:

All JSON messages are valid YAML. Therefore, in addition to using YAML request bodies for Server-Side Apply requests, you can also use JSON request bodies, as they are also valid YAML.

Reply:
> All JSON messages are valid YAML. Therefore, in addition to using YAML request bodies for Server-Side Apply requests, you can also use JSON request bodies, as they are also valid YAML.

You could try that. Actually, the existing text is factually wrong.  JSON allows tabs as whitespace; YAML does not. Something to bear in mind.

You can convert all JSON to valid YAML by replacing all ASCII tab characters with the appropriate number of space (`0x20`) characters. If you do that, the resulting YAML will also be valid JSON.

Reply:
[YAML](https://yaml.org/spec/1.2.2/) is a superset of JSON, and any compliant YAML parser should be able to parse JSON. There’s no need to convert a JSON document before passing it to a YAML parser.

Reply:
Can you cite a source for that assertion [that the existing docs are correct and that YAML is a pure superset of JSON]? I checked against the official grammars.

Reply:
The documentation is **incorrect**, as it states the opposite. SSA request bodies (`application/apply-patch+yaml` content type) are expected to be in YAML. However,  since a YAML parser can also parse JSON (YAML is a superset of JSON), you can use JSON as well as YAML for request bodies.
The existing documentation states:

> Some clients specify Server-Side Apply requests using YAML request bodies that are also valid JSON.

This should be corrected to:

Some clients specify Server-Side Apply requests using JSON request bodies that are also valid YAML.

Reply:
I think no one is wrong or right here, but if there's confusion, we should probably update it.

> The existing documentation states:
> 
> > Some clients specify Server-Side Apply requests using YAML request bodies that are also valid JSON.

I don't think that's wrong, I could slightly rephrase as:
> Some clients specify Server-Side Apply requests using YAML request bodies, and they happen to also be valid JSON (which they can be), but that's irrelevant, since it's still YAML at the end of the day. So in other word, however you format your YAML, in the form of json or not, it doesn't matter.

Reply:
> Some clients specify Server-Side Apply requests using YAML request bodies that are also valid JSON.

From this statement, it can be inferred that all YAML are valid JSON. At least, this is my initial interpretation. While this is true for some YAML documents, it is not true for all YAML. Furthermore, since the content type for SSA requests is `application/apply-patch+yaml`, the request body is expected to be in YAML. Given that YAML is a superset of JSON, I thought there was a typo in the statement and you meant to say that some clients use **JSON** for request bodies, which is also valid **YAML**.